pkg/controller/controller_utils_test.go

```diff
@@ -597,29 +694,42 @@ func TestSortingActivePodsWithRanks(t *testing.T) {
 			if podsWithRanks.Less(1, 0) {
 				t.Errorf("expected pod %q with rank %v not to be less than %v with rank %v", podsWithRanks.Pods[1].Name, podsWithRanks.Rank[1], podsWithRanks.Pods[0].Name, podsWithRanks.Rank[0])
 			}
-		})
-	}
+		}
+	})
 }
 
 func TestActiveReplicaSetsFiltering(t *testing.T) {
-	var replicaSets []*apps.ReplicaSet
-	replicaSets = append(replicaSets, newReplicaSet("zero", 0))
-	replicaSets = append(replicaSets, nil)
-	replicaSets = append(replicaSets, newReplicaSet("foo", 1))
-	replicaSets = append(replicaSets, newReplicaSet("bar", 2))
-	expectedNames := sets.NewString()
-	for _, rs := range replicaSets[2:] {
-		expectedNames.Insert(rs.Name)
-	}
 
-	got := FilterActiveReplicaSets(replicaSets)
-	gotNames := sets.NewString()
-	for _, rs := range got {
-		gotNames.Insert(rs.Name)
+	tests := []struct {
+		name          string
+		replicaSets   []*apps.ReplicaSet
+		expectedNames []string
+	}{
+		{
+			name: "Filters active replica sets",
+			replicaSets: []*apps.ReplicaSet{
+				newReplicaSet("zero", 0),
+				nil,
+				newReplicaSet("foo", 1),
+				newReplicaSet("bar", 2),
+			},
+			expectedNames: []string{"foo", "bar"},
+		},
 	}
 
-	if diff := cmp.Diff(expectedNames.List(), gotNames.List()); diff != "" {
-		t.Errorf("Active replica set names (-want,+got):\n%s", diff)
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got := FilterActiveReplicaSets(test.replicaSets)
+
+			gotNames := make([]string, len(got))
```

Comment:
Do you mean replacing the explicit expectation 

`expectedNames: []string{"foo", "bar"},`

by the code that was before? 

```
var replicaSets []*apps.ReplicaSet
replicaSets = append(replicaSets, newReplicaSet("zero", 0))
replicaSets = append(replicaSets, nil)
replicaSets = append(replicaSets, newReplicaSet("foo", 1))
replicaSets = append(replicaSets, newReplicaSet("bar", 2))
expectedNames := sets.NewString()
for _, rs := range replicaSets[2:] {
	expectedNames.Insert(rs.Name)
}
```

Reply:
Sorry, I think I don't understand completely. Do you mean replacing the explicit expectations
 
```
expectedNames: []string{"foo", "bar"},
```

with the original code?
```
var replicaSets []*apps.ReplicaSet
	replicaSets = append(replicaSets, newReplicaSet("zero", 0))
	replicaSets = append(replicaSets, nil)
	replicaSets = append(replicaSets, newReplicaSet("foo", 1))
	replicaSets = append(replicaSets, newReplicaSet("bar", 2))
	expectedNames := sets.NewString()
	for _, rs := range replicaSets[2:] {
		expectedNames.Insert(rs.Name)
	}
```

Regarding the diff, I do it [below](https://github.com/kaisoz/kubernetes/blob/02aca97542f83229b15b7b1e1c4aeb017e7e7172/pkg/controller/controller_utils_test.go#L472)
```
if diff := cmp.Diff(test.expectedNames, gotNames.List()); diff != "" {
				t.Errorf("Active pod names (-want,+got):\n%s", diff)
			}
```
and it's practically the same as it was before. All I do is replace sets with slices (because specified sets are now [deprecated](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets), and explicitly define the expectations instead of generating them dynamically from the replicaSet slice.

Reply:
Gotcha! I slightly refactored the previous solution and I "kind of" understand why it was written like that. 

Since in Golang you can't reference a struct member from another struct member at definition time, one way to "anchor" the expected ReplicaSets to the input ReplicaSets it's by name (which is supposed to be unique and representative). I can't have an array of `expectedReplicaSets` created with the `newReplicaSet` function because [they'd have different `UUID`s](https://github.com/kaisoz/kubernetes/blob/02aca97542f83229b15b7b1e1c4aeb017e7e7172/pkg/controller/controller_utils_test.go#L138) and therefore fail the test. 

I could maybe modify the `newReplicaSet` function to inject the UUID (which would be part of the test struct declaration) so that I have total control over what it generates... Other solutions that occur to me would also require pre/post-processing before calling the `FilterActiveReplicaSets` function and TBH I'm not sure if they are better than this...

WDYT? Do you have any suggestions?

Reply:
I think defining replicaset or passing in a UID would be a good idea.